pkg/storages/azure/folder.go

```diff
@@ -124,7 +124,7 @@ func ConfigureFolder(prefix string, settings map[string]string) (storage.Folder,
 
 	var containerUrlString string
 	var containerClient azblob.ContainerClient
-	if usingToken {
+	if usingToken && !usingAccessKey {
```

Comment:
Can we introduce some type to define the chosen auth method to make this logic more straightforward?

For example:

```
type AzureAuthType string

const ( 
AzureTokenAuth AzureAuthType = "AzureTokenAuth"
AzureAccessKeyAuth AzureAuthType = "AzureAccessKeyAuth"
)
```

So we can separate the auth logic by the `AzureAuthType`.

Reply:
makes sense, pushed a change